### PR TITLE
Feature/drawer example

### PR DESCRIPTION
**Pull request by ektasawant**
Fixes PXBLUE-2296 .

#### Changes proposed in this Pull Request:
- [X] Design Extended header with gradient effect 
- [X] Create drawer  component to match design
- [X] Change drawer width to 292px from 270px as per design
- [X] Create drawer  component to match design
- [X] Close drawer when user clicks on close icon/ black overlay
- [X] Test Mobile view

#### Screenshots:

Desktop_Drawer

image

#### To Test:
- Local test URL http://localhost:3000/in-a-drawer

### CODE REVIEW

**Comment by emclaug2** [comment]
Couple things here:

- Could you change the Assets icon to be this icon: https://pxblue.github.io/style/iconography?icon=Device&isMaterial=false
- Should the bottom-aligned nav items be selected onclick @huayunh ?
- The X button whenever clicked, the affected ripple area seems a little large to me.

Overall looking really good visually.

File: src/pages/account-menu/in-a-drawer/Drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    const ExtendedDrawerHeader = (): JSX.Element => (
+        <div className={classes.extendedHeader}>
+            <div className={classes.header}>
+                <Avatar alt="Chima Thabani" src={avatarImage} classes={{ root: classes.avatarSize }} />
+                <IconButton
+                    data-cy="toolbar-menu"
+                    color={'inherit'}
+                    edge={'end'}
+                    classes={{ edgeEnd: classes.closeIcon }}
+                    onClick={drawerToggler}
```
Comment (joebochill) [inline src/pages/account-menu/in-a-drawer/Drawer.tsx:112]: I would call this `toggleDrawer` just for the sake of keeping functions in the verb form

File: src/pages/account-menu/in-a-drawer/Drawer.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        },
+        {
+            title: 'Assets',
+            itemID: '5',
+            icon: <WebAssetIcon />,
+            onClick: (): void => setSelected('5'),
+        },
+    ];
+
+    const ExtendedDrawerHeader = (): JSX.Element => (
```
Comment (joebochill) [inline src/pages/account-menu/in-a-drawer/Drawer.tsx:103]: Maybe call this something like DrawerHeaderContent...ExtendedDrawerHeader makes me think this is a component that extends the drawer header component, but it's just the custom title piece that goes inside it.

File: src/pages/account-menu/in-a-drawer/Drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+};
+
+const avatarImage = require('../../../assets/avatar_40.png').default;
+
+export const PxbDrawer = (props: DrawerProps): JSX.Element => {
+    const { open, drawerToggler } = props;
+    const theme = useTheme();
+    const classes = useStyles(theme);
+    const variant = 'temporary';
+    const [selected, setSelected] = useState('1');
```
Comment (joebochill) [inline src/pages/account-menu/in-a-drawer/Drawer.tsx:68]: I would be totally fine with not making any of the drawer selections functional...they're not the main focus of the example and they could be distracting. Just a thought.

**Review by joebochill — COMMENTED** [review]
This looks really nice!

**Comment by joebochill** [comment]
> * The X button whenever clicked, the affected ripple area seems a little large to me.

I'm pretty sure that's the standard 40x40 ripple area used for icon buttons. It looks to be the same size as the ones on the doc-it drawer.

File: src/pages/account-menu/in-a-drawer/Drawer.tsx (hunk 3 above)
Comment (ektasawant) [reply src/pages/account-menu/in-a-drawer/Drawer.tsx:68]: For showing below spec (active menu item's background shape should be round) I need to select menu item
image

**Comment by ektasawant** [comment]
> Couple things here:
> 
> * Could you change the Assets icon to be this icon: https://pxblue.github.io/style/iconography?icon=Device&isMaterial=false => **Done**
> * Should the bottom-aligned nav items be selected onclick @huayunh ?
> * The X button whenever clicked, the affected ripple area seems a little large to me.
> 
> Overall looking really good visually.

**Review by joebochill — APPROVED** [review]

**Pull request merged**